### DISCUSSION TITLE
【Help】 meshlab Boolean operation problem consultation

### DISCUSSION THREAD

**Discussion by ning2510** [cnr-isti-vclab/meshlab#1397] [Q&A]
I've recently been using boolean operations in libigl, but the calculations are less ef

ficient for complex 3D objects. But when I use meshlab for boolean operations, the time can be 2~3 times faster. After that, I took a brief look at the source code of the Boolean operation module in meshlab, and also called the libigl library, but why is the efficiency much faster than calling libigl myself?

It can be seen that meshlab also calls the Boolean operation in libigl through the igl::copyleft::cgal::mesh_boolean() interface.
![](https://github.com/cnr-isti-vclab/meshlab/assets/48829770/6e73c030-977c-4ec6-8369-2721fae97dbe)
I tried to test with 3D objects with different numbers of triangles, and the speed of meshlab is about 2 or 3 times faster than my own code.

**Comment by alemuntoni** [comment]
That's probably related to compilation flags, or floating point precision (MeshLab 2022.02 uses floats, 2022.02d uses double). MeshLab does nothing magic on boolean operations, it just calls igl.

**Reply by ning2510** [reply]
Thanks !
